e2e/testcases/composition_test.go

```diff
@@ -373,10 +373,10 @@ func waitForSync(nt *nomostest.NT, sha1Func nomostest.Sha1Func, objs ...client.O
 func validateStatusCurrent(nt *nomostest.NT, objs ...client.Object) {
 	tg := taskgroup.New()
 	for _, obj := range objs {
+		nn := client.ObjectKeyFromObject(obj)
 		tg.Go(func() error {
 			return nomostest.WatchForCurrentStatus(
-				nt, obj.GetObjectKind().GroupVersionKind(), obj.GetName(), obj.GetNamespace(),
-			)
+				nt, obj.GetObjectKind().GroupVersionKind(), nn.Name, nn.Namespace)
```

Comment:
Does `obj.GetObjectKind().GroupVersionKind()` yield the same result as `kinds.Lookup(obj, nt.scheme)`?

Reply:
Only if the GVK is populated on the object. For Unstructured objects GVK is always populated, but for typed objects it’s only sometimes populated, depending on source. The lookup uses the GVK if set, otherwise looks up the GVK using the Scheme’s type map.

Reply:
It looks like in the case of validateStatusCurrent, it's always being called with objects from a Repository.GetAll, which delegates to Repository.GetAll, which parses to Unstructured and then converts to typed before returning, but the method it uses to convert to typed (runtime.DefaultUnstructuredConverter.FromUnstructured) leaves the GVK on the typed object.

I'll change the func to use Lookup, just in case that changes in the future, but there's a lot of tests relying on that implementation detail today.